DXRModules/DeusEx/Classes/DXRFixup.uc

```diff
@@ -797,6 +806,38 @@ static function FixConversationAddNote(Conversation c, string textSnippet)
     }
 }
 
-defaultproperties
+function SetLampState(#var(prefix)Lamp lmp, bool turnOn) // could maybe go in DXRLamp if it existed
+{
+    if (turnOn) {
+        lmp.bOn = True;
+        lmp.LightType = LT_Steady;
+        // lmp.PlaySound(sound'Switch4ClickOn');
+        lmp.bUnlit = True;
+        lmp.ScaleGlow = 3.0;
+    } else if (lmp.bOn) {
+        lmp.bOn = False;
+        lmp.LightType = LT_None;
+        // lmp.PlaySound(sound'Switch4ClickOff');
+        lmp.bUnlit = False;
+        lmp.ResetScaleGlow();
+    }
+}
+
+// the Plane struct is repurposed here to describe a disk orthogonal to the xy-plane
```

Comment:
Since you're using RadiusActors, the plane isn't being used to define a circle, it's defining a sphere in this case.

Based on what this comment describes, if I didn't read the contents of the function, I would assume you were doing a height check as well to define a cylinder

Reply:
Now that I better understand what RadiusActors does, the comment will be fixed.

Reply:
The fact that you got confused by your own usage of the Plane struct is a good indication that it should not be used for this purpose. RadiusActors uses a vector for location and a separate float for the radius, we can do the same thing as they do, it's the proper convention.

Reply:
I was confused about RadiusActor. That wouldn't have been any better with separate Vector and radius parameters.